tests/checkout/unit_test.go

```diff
@@ -0,0 +1,244 @@
+package checkout
+
+import (
+	"context"
+	"github.com/adyen/adyen-go-api-library/v8/src/adyen"
+	"github.com/adyen/adyen-go-api-library/v8/src/checkout"
+	"github.com/adyen/adyen-go-api-library/v8/src/common"
+	"github.com/adyen/adyen-go-api-library/v8/tests"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestCheckout(t *testing.T) {
+	client := adyen.NewClient(&common.Config{
+		ApiKey:      "YOUR_ADYEN_API_KEY",
+		Environment: "TEST",
+	})
+	merchantAccount := "YOUR_MERCHANT_ACCOUNT"
+
+	mux := http.NewServeMux()
+
+	mockResponse := tests.MockResponse(t, mux)
+	mockOk := func(method, endpoint, fixture string) {
+		mockResponse(http.StatusOK, method, endpoint, fixture)
+	}
+	mockCreated := func(method, endpoint, fixture string) {
+		mockResponse(http.StatusCreated, method, endpoint, fixture)
+	}
+
+	// Success cases
+	mockOk("POST", "/payments", "checkout/authorised_card.json")
+	mockOk("POST", "/paymentMethods", "checkout/payment_methods.json")
+	mockOk("POST", "/paymentMethods/balance", "checkout/gift_card_balance.json")
+	mockOk("POST", "/orders", "checkout/order_created.json")
+	mockOk("GET PATCH", "/paymentLinks/PL61C53A8B97E6915A", "checkout/payment_link.json")
+	mockCreated("POST", "/sessions", "checkout/session_created.json")
+
+	mux.HandleFunc("/orders/cancel", func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, "POST", r.Method)
+		w.Header().Set("Content-Type", "application/json")
+		io.WriteString(w, `{
+		  "pspReference": "8816178914079738",
+		  "resultCode": "Received"
+		}`)
+	})
+
+	mockServer := httptest.NewServer(mux)
+	defer mockServer.Close()
+	client.Checkout().PaymentsApi.BasePath = func() string { return mockServer.URL }
+	service := client.Checkout()
+
+	t.Run("Configuration", func(t *testing.T) {
+		liveClient := adyen.NewClient(&common.Config{
+			ApiKey:                "LIVE_API_KEY",
+			Environment:           common.LiveEnv,
+			LiveEndpointURLPrefix: "abc123",
+			Debug:                 false,
+		})
+		require.Equal(t, "https://abc123-checkout-live.adyenpayments.com/checkout/v70", liveClient.Checkout().PaymentsApi.BasePath())
```

Comment:
can we use `CheckoutAPIVersion` or a constant, instead of hardcoding `v70`? 
It will be easier to update for future versions 👍